src/Angor/Client/Services/IconService.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+
+namespace Angor.Client.Services
+{
+    public class IconService
+    {
+        private readonly Dictionary<string, string> _icons = new Dictionary<string, string>();
+
+        public IconService()
+        {
+            LoadIcons();
+        }
+
+        public Task<string> GetIcon(string iconName, int width, int height, string color)
+        {
+            if (_icons.TryGetValue(iconName, out var svgContent))
```

Comment:
the cache should be on several parameters (iconName-width-height-cololr otherwise you will get inv=correct values

Reply:
Icons are not cached and are always called with code. They are embedded in the app